FlowCrypt/Controllers/Compose/ComposeViewController.swift

```diff
@@ -1060,50 +1077,49 @@ extension ComposeViewController {
             UIAlertAction(
                 title: "files_picking_camera_input_source".localized,
                 style: .default,
-                handler: { [weak self] _ in
-                    guard let self = self else { return }
-                    self.photosManager.takePhoto(from: self)
-                        .sinkFuture(
-                            receiveValue: {},
-                            receiveError: { _ in
-                                self.showNoAccessToCameraAlert()
-                            }
-                        )
-                        .store(in: &self.cancellable)
-                }
+                handler: { [weak self] _ in self?.takePhoto() }
             )
         )
         alert.addAction(
             UIAlertAction(
                 title: "files_picking_photo_library_source".localized,
                 style: .default,
-                handler: { [weak self] _ in
-                    guard let self = self else { return }
-                    self.photosManager.selectPhoto(from: self)
-                        .sinkFuture(
-                            receiveValue: {},
-                            receiveError: { _ in
-                                self.showNoAccessToPhotosAlert()
-                            }
-                        )
-                        .store(in: &self.cancellable)
-                }
+                handler: { [weak self] _ in self?.selectPhoto() }
             )
         )
         alert.addAction(
             UIAlertAction(
                 title: "files_picking_files_source".localized,
                 style: .default,
-                handler: { [weak self] _ in
-                    guard let self = self else { return }
-                    self.filesManager.selectFromFilesApp(from: self)
-                }
+                handler: { [weak self] _ in self?.selectFromFilesApp() }
             )
         )
         alert.addAction(UIAlertAction(title: "cancel".localized, style: .cancel))
         present(alert, animated: true, completion: nil)
     }
 
+    private func takePhoto() {
+        Task {
+            do {
+                try await photosManager.takePhoto(from: self)
+            } catch {
+                showNoAccessToCameraAlert()
+            }
+        }
+    }
+
+    private func selectPhoto() {
+        Task {
+            await photosManager.selectPhoto(from: self)
+        }
+    }
+
+    private func selectFromFilesApp() {
+        Task {
+            await filesManager.selectFromFilesApp(from: self)
+        }
+    }
```

Comment:
Here in `takePhoto` we should distinguish the exact "missing access" photo before running `showNoAccessToCameraAlert`. Else a regular error modal.

`selectPhoto` and `selectFromFilesApp` should have errors handled with a modal?

Let's look at this in the future.

Reply:
We don't have errors in `selectPhoto` and `selectFromFilesApp`, so I don't use `throws`.